arcgis-ios-sdk-samples/Maps/Create and save a map/SaveAsViewController.swift

```diff
@@ -38,12 +45,51 @@ class SaveAsViewController: UITableViewController {
             presentAlert(message: "Please enter a title.")
             return
         }
-        
+        // Get the tags from the text field.
         let tags = tagsTextField.text?
             .components(separatedBy: ",")
             .map { $0.trimmingCharacters(in: .whitespacesAndNewlines) } ?? []
-        
+        // Get the item description from the text field.
         let itemDescription = descriptionTextField.text ?? ""
-        delegate?.saveAsViewController(self, didInitiateSaveWithTitle: title, tags: tags, itemDescription: itemDescription)
+        
+        delegate?.saveAsViewController(self, didInitiateSaveWithTitle: title, tags: tags, itemDescription: itemDescription, folder: selectedFolder)
+    }
+    
+    /// Present the list of folders to choose from.
+    func showFolderOptions() {
+        // Set index for the default option.
+        let selectedIndex = portalFolders.firstIndex { $0 == selectedFolder } ?? portalFolders.count
+        // The titles of the portal folders and a "No folder" option.
+        let folderTitles = portalFolders.map { $0.title! } + ["No folder"]
+        // Prepare the options table view controller and handle selection.
+        let optionsViewController = OptionsTableViewController(labels: folderTitles, selectedIndex: selectedIndex) { [weak self] newIndex in
+            guard let self = self else { return }
+            // If "No folder" was selected, set selectedFolder to nil.
+            if newIndex == self.portalFolders.count {
```

Comment:
It would be great if the user could just tap on a selected row to unselect it. I see that `OptionsTableViewController` doesn't currently support that, but it would be easy enough to add. So as not to change the behavior of existing samples, you could add an `allowsEmptySelection` property (with a default of `false`) to `OptionsTableViewController` and only allow deselection when that property is set to `true`.

Reply:
Are you suggesting that I make this change in `OptionsTableViewController.swift` in the Cocoa repo?

Reply:
This https://github.com/Esri/arcgis-runtime-samples-ios/blob/v.next/arcgis-ios-sdk-samples/Content%20Display%20Logic/Controllers/OptionsTableViewController.swift

Reply:
Update: this issue does not impede the user's workflow thus, will be handled in a separate PR `common-samples/issues/2850` after the release. This decision is was made to prioritize the release and to ensure that we'll release according to schedule. Please continue reviewing the sample in its current state without this feature for now. Thank you!